delivery_purchase/models/delivery_carrier.py

```diff
@@ -159,4 +164,27 @@ def _purchase_get_price_available(self, order):
             order.company_id,
             order.date_order or fields.Date.today(),
         )
+        return weight, volume, quantity, total
+
+    def _purchase_get_price_available(self, order):
+        self.ensure_one()
+        self = self.sudo()
+        order = order.sudo()
+        lines = order.order_line
+        qty_field = "product_uom_qty"
+        weight, volume, quantity, total = self._get_delivery_data(
+            order, lines, qty_field
+        )
```

Comment:
Don't put variables with only one usage (the same for the rest of the code):

```suggestion
        order = order.sudo()
        weight, volume, quantity, total = self._get_delivery_data(
            order, order.order_line, "product_uom_qty"
        )
```

Do the code more readable not having to follow which is the value of such local variables.